### PR TITLE
[SPARK-16245] [ML] model loading backward compatibility for ml.feature.PCA

### PR DESCRIPTION
**Pull request by yanboliang**
## What changes were proposed in this pull request?

model loading backward compatibility for ml.feature.PCA.
## How was this patch tested?

existing ut and manual test for loading models saved by Spark 1.6.

### CODE REVIEW

**Comment by yanboliang** [comment]
cc @hhbyyh @mengxr

File: mllib/src/main/scala/org/apache/spark/ml/feature/PCA.scala [hunk 1]
```diff
@@ -206,24 +206,21 @@ object PCAModel extends MLReadable[PCAModel] {
...
             .select("pc", "explainedVariance")
             .head()
         new PCAModel(metadata.uid, pc, explainedVariance)
       } else {
-        val Row(pc: DenseMatrix) = sparkSession.read.parquet(dataPath).select("pc").head()
-        new PCAModel(metadata.uid, pc, Vectors.dense(Array.empty[Double]).asInstanceOf[DenseVector])
+        // explainedVariance field is not present and we use the old matrix in Spark <= 2.0
+        val Row(pc: OldDenseMatrix) = sparkSession.read.parquet(dataPath).select("pc").head()
+        new PCAModel(metadata.uid, pc.asML,
+          Vectors.dense(Array.empty[Double]).asInstanceOf[DenseVector])
```
Comment (yanboliang) [inline mllib/src/main/scala/org/apache/spark/ml/feature/PCA.scala]: Here we combine the ```explainedVariance``` field issue and the old matrix issue together to handle backward compatibility.

**Comment by hhbyyh** [comment]
LGTM

**Comment by mengxr** [comment]
Merged into master and branch-2.0. Thanks!

**Pull request closed without merging**